tests/closure2.nu

```diff
@@ -1,3 +1,6 @@
+let a = 10
+let b = 20
+
```

Comment:
In this example we will rewind with the `RollbackPoint` so if the first expresssion would declare variables/scopes, this would either also have to rewind the scope frames or it would duplicate the declarations.
Alternatively we move the whole scope declaration and lookup logic to its own pass?